rust/lance/src/dataset/scanner.rs

```diff
@@ -5009,17 +5009,20 @@ mod test {
         assert!(filtered_scan_bytes < full_scan_bytes);
 
         // Now do a scan with pushdown, the benefit should be even greater
-        let start_bytes = get_bytes();
-        dataset
-            .scan()
-            .filter("not_indexed = 50")
-            .unwrap()
-            .try_into_batch()
-            .await
-            .unwrap();
-        let pushdown_scan_bytes = get_bytes() - start_bytes;
+        // Pushdown only works with the legacy format for now.
+        if data_storage_version == LanceFileVersion::Legacy {
+            let start_bytes = get_bytes();
+            dataset
+                .scan()
+                .filter("not_indexed = 50")
+                .unwrap()
+                .try_into_batch()
+                .await
+                .unwrap();
+            let pushdown_scan_bytes = get_bytes() - start_bytes;
 
-        assert!(pushdown_scan_bytes < filtered_scan_bytes);
+            assert!(pushdown_scan_bytes < filtered_scan_bytes);
+        }
```

Comment:
I found that once I started caching the index metadata, there wasn't a different in bytes read for the stable format. It seems like this was only passing because the first call was calling `load_indices()`, and the second call had the indices cached. Does this seem right @westonpace?

Reply:
Yes, that seems correct.  Currently, pushdown only gives significant benefits with the legacy format.  The plan for the stable format is to add support for zone map / bloom filters as a secondary (best-effort) index.  Now that we have ngram we've paved out the path for these best-effort indices pretty well.